src/compiler/types.ts

```diff
@@ -700,6 +700,7 @@ namespace ts {
         Unique = 3,                         // Unique name based on the 'text' property.
         Node = 4,                           // Unique name based on the node in the 'original' property.
         OptimisticUnique = 5,               // Unique name based on the 'text' property, first instance won't use '_#' if there's no conflict
+        FileLevel = 6,                      // Optimistic unique name based on the 'text' property, first instance won't use '_#' if there's no conflict, using only the file identifiers list

```

Comment:
Both `FileLevel` and `OptimisticUnique` seem more like flags than kinds, since they just augment the behavior of `makeUniqueName` in emitter.ts.